### PR TITLE
boards: shields: Add support for Mikroe BLE TINY Click shield

### PR DESCRIPTION
**Pull request by iandmorris**
Add support for the Mikroe BLE TINY Click shield that contains the Renesas DA14531 Bluetooth LE Module.

### CODE REVIEW

File: boards/shields/mikroe_ble_tiny_click/doc/ble-tiny-click.jpg
Comment (kartben) [inline boards/shields/mikroe_ble_tiny_click/doc/ble-tiny-click.jpg:1]: please use webp (smaller file size, supports transparent background) and crop excess white space

you may directly use https://gist.githubusercontent.com/kartben/493373edfcb0940a7c9c0a230fd2e821/raw/7f727942d35c41ce7aff579d7d544d44c46a3d96/ble_tiny_click.webp

File: boards/shields/mikroe_ble_tiny_click/doc/da14531-hci-binary.jpg
Comment (kartben) [inline boards/shields/mikroe_ble_tiny_click/doc/da14531-hci-binary.jpg:1]: please convert to webp

https://gist.githubusercontent.com/kartben/493373edfcb0940a7c9c0a230fd2e821/raw/84844690fdca3f52bbd58590b6c696c5a22b58bb/da14531-hci-binary.webp

**Review by kartben — CHANGES_REQUESTED** [review]
very nice! Thanks for the contribution

File: boards/shields/mikroe_ble_tiny_click/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+This shield can only be used with a board that provides a mikroBUS |trade|
+socket and defines the ``mikrobus_serial`` node label (see :ref:`shields`
+for more details).
+
+.. note::
+
+   The reset input on the DA14531 Module is active high and is connected to
+   the RST pin on the mikroBUS socket. On many host boards this RST pin is
+   connected to a system reset signal that is active low. This results in the
+   host system unintentinally holding the DA14531 Module in reset. This issue
```
Comment (kartben) [inline boards/shields/mikroe_ble_tiny_click/doc/index.rst:30]:
```suggestion
   host system unintentionally holding the DA14531 Module in reset. This issue
```

File: boards/shields/mikroe_ble_tiny_click/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+   can be overcome by removing resistor R3 on the BLE TINY Click board, see
+   the `BLE TINY Click Schematic`_ for further details.
+
+The DA14531 Module contained on the shield must programmed with a binary file
```
Comment (kartben) [inline boards/shields/mikroe_ble_tiny_click/doc/index.rst:34]:
```suggestion
The DA14531 Module contained on the shield must be programmed with a binary file
```

File: boards/shields/mikroe_ble_tiny_click/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+that supports the HCI interface over UART, without hardware flow control as
+these signals are not support on the Click footprint.
```
Comment (kartben) [inline boards/shields/mikroe_ble_tiny_click/doc/index.rst:36]:
```suggestion
these signals are not supported on the Click footprint.
```

**Review by kartben — COMMENTED** [review]
spotted some minor typos, apologies that I missed them during earlier pass
LGTM otherwise, thanks again!

**Comment by kartben** [comment]
@iandmorris please also rebase to address unrelated CI failure

**Review by kartben — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**